store/sqlstore/user_store.go

```diff
@@ -1361,3 +1361,466 @@ func (us SqlUserStore) InferSystemInstallDate() store.StoreChannel {
+
+				WHERE (TargetInteraction.Interaction_Type = 'DirectMessage' -- direct message from someone other than current user in group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND ChannelMembers.UserId = :uid)
+
+				OR  (TargetInteraction.Interaction_Type = 'DirectMessage' -- direct message from current user in group channel
+				    AND TargetInteraction.UserId = :uid
+				    AND ChannelMembers.UserId != :uid)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, not written by current user, to a post that the current user IS an author of, in a group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a reply)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, not written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId != :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id = :uid) -- just want one user returned from channel members (the current user) (will record as a DM)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS NOT an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id != :uid
+				        AND ChannelMembersUsers.Id != :uid)
+
+				OR (TargetInteraction.Interaction_Type = 'Reply' -- reply, written by current user, to a post that the current user IS an author of, in a group channel
+				        AND TargetInteraction.UserId = :uid
+				        AND Parent_Post_Author_Id = :uid
+				        AND ChannelMembersUsers.Id != :uid)
+				)
+			) AS UserInteractions
+
+			WHERE Username IN (SELECT Username -- all interactions must take place with users on your team
+			                          FROM Users
+
+			                            JOIN TeamMembers
+			                            ON TeamMembers.TeamID = :teamId
+			                            AND TeamMembers.UserId = Users.Id
+			                        )
+			`
+			var userInteractions []*model.UserInteraction
+
+			if _, err :=  us.GetReplica().Select(&userInteractions, query, map[string]interface{}{"uid": uid, "teamId": teamId}); err != nil {
+				result.Err = model.NewAppError("SqlUserStore.GetUserInteractions", "store.sql_post.ge_user_interactions.app_error", nil, err.Error(), http.StatusInternalServerError)
+			}
+
+			result.Data = userInteractions
+	})
+}
+
+func (us *SqlUserStore) GetUserLearningGroups(uid string, teamId string, learningGroupPrefixes []*model.LearningGroupPrefix) store.StoreChannel {
+	// For every learning group prefix for this team, use this case statement
+	// to record a name for this channel(team), minus the prefix (ex. plg-30 => 30)
+	prefixReplaceQuery := ""
```

Comment:
It looks like there's some boiler-plate here. This makes me even more sure that query builders would be helpful in making this code more readable.

Reply:
I agree that this SQL query is very large and would be better broken into smaller parts.
A query builder might be a good way to go, but I'm loathe to figure out how one correctly adds a package (and figure out the right query builder package to use) in a go project at this time.
What I think would be the compromise is to note this in in a TODO type comment at the top of the query, or in the function definition as something that should be addressed either explicitly or the next time a story requires a change to this function.